src/main/kotlin/no/nav/familie/ef/sak/iverksett/oppgaveforbarn/BarnFyllerÅrOppfølgingsoppgaveService.kt

```diff
@@ -52,26 +54,64 @@ class BarnFyllerÅrOppfølgingsoppgaveService(
-        return barnTilUtplukkForOppgave.filter { it.fødselsnummerBarn != null } +
-            finnFødselsnummerTilTerminbarn(barnUtenFødselsnummer)
+        val barnMedFødselsnummer = barnTilUtplukkForOppgave.filter { it.fødselsnummerBarn != null }
+        val barnUtenIdentOppdatertMedIdentFraPdl = finnFødselsnummerTilTerminbarn(barnUtenFødselsnummer)
+        return barnMedFødselsnummer + barnUtenIdentOppdatertMedIdentFraPdl
+    }
+
+    private fun hentFødselsdatoFraGrunnlagsdata(barn: BarnTilUtplukkForOppgave): LocalDate? {
+        val grunnlagsdata = grunnlagsdataService.hentGrunnlagsdata(barn.behandlingId)
+        val barnAvGrunnlagsdata = grunnlagsdata.grunnlagsdata.barn.filter { it.personIdent == barn.fødselsnummerBarn }
+        return barnAvGrunnlagsdata
+            .first()
+            .fødsel
+            .first()
+            .fødselsdato
     }
 
     private fun finnFødselsnummerTilTerminbarn(barnMedTermindato: List<BarnTilUtplukkForOppgave>): List<BarnTilUtplukkForOppgave> {
-        val pdlPersonMedForelderBarnRelasjon =
-            personService.hentPersonForelderBarnRelasjon(barnMedTermindato.map { it.fødselsnummerSøker })
+        val forelderBarn: Map<ForelderIdentDto, List<BarnMedFødselsdatoDto>> = hentForelderMedBarnFor(barnMedTermindato)
 
         return barnMedTermindato.map { barn ->
-            val termindato = barn.termindatoBarn!!
-            val søkerPdlData =
-                pdlPersonMedForelderBarnRelasjon[barn.fødselsnummerSøker]
-                    ?: error("Finner ikke pdldata for søker=${barn.fødselsnummerSøker}")
-            val forelderBarnRelasjoner = søkerPdlData.forelderBarnRelasjon.mapNotNull { it.relatertPersonsIdent }
-            val besteMatch = finnBesteMatchPåFødselsnummerForTermindato(forelderBarnRelasjoner, termindato)
-
+            val alleBarnaTilForelder = forelderBarn[ForelderIdentDto(barn.fødselsnummerSøker)] ?: emptyList()
+            val besteMatch = finnBesteMatchPåFødselsnummerForTermindato(alleBarnaTilForelder, barn.termindatoBarn ?: error("Termindato er null"))
             barn.copy(fødselsnummerBarn = besteMatch)
         }
     }
 
+    private fun hentForelderMedBarnFor(barnMedTermindato: List<BarnTilUtplukkForOppgave>): Map<ForelderIdentDto, List<BarnMedFødselsdatoDto>> {
+        val forelderIdentMedBarn: Map<ForelderIdentDto, List<String>> = hentForeldreMedBarn(barnMedTermindato)
+        logger.info("Antall foreldre vi prøver å matche: ${forelderIdentMedBarn.keys.size}")
+        val alleBarn = forelderIdentMedBarn.values.flatten()
+        val barnMedFødselsdato: Map<String, LocalDate?> = hentBarnFødselsdatoer(alleBarn)
+        logger.info("Antall barn hentet ut med fødselsdatoer: ${alleBarn.size}")
+        return forelderIdentMedBarn.map { it.key to it.value.map { barneIdent -> BarnMedFødselsdatoDto(barneIdent, barnMedFødselsdato[it.key.ident]) } }.toMap()
```

Comment:
```suggestion
        return forelderIdentMedBarn.map { forelder -> forelder.key to forelder.value.map { barneIdent -> BarnMedFødselsdatoDto(barneIdent, barnMedFødselsdato[barneIdent]) } }.toMap()
```

Reply:
`it` refererer til forelder, og `it.key.ident` er i dette tilfellet forelders ident. 

Skulle gjerne hatt en test som passer på at vi får "treff" på riktig liste her.

Reply:
Bra observert!